pkg/openwsn/doc.txt

```diff
@@ -0,0 +1,218 @@
+ *
+ * The OpenWSN software provides different hooks all over the stack to toggle
+ * different LEDs as well as debug pins to examine state and scheduling of a node.
+ * Default configuration files are provided for both. The LED configuration maps to
+ * RIOTs `LEDX_PIN` definitions, if available. On Nucleo boards LED0 line is
+ * shared SPI, so is not used.
+ * The default configuration can be overwritten by setting `OPENWSN_LEDPINS_DEFAULT`
+ * in the form of `ledpins_config_t`. The debugpins work similarly by setting
+ * `OPENWSN_DEBUGPINS_BOARD` in the form of `debugpins_config_t`.
+ *
+ * The default configuration maps to OpenWSN reference hardware `openmote-b`.
+ *
+ * ## Testing and debugging
+ *
+ * List of some items which are helpful to explore the functionality of OpenWSN:
+ *
+ * - LED pins and debug pins as mentioned above in combination with a logic analyzer.
+ * The expected behavior is described in:
+ * [OpenWSN wiki](https://openwsn.atlassian.net/wiki/spaces/OW/pages/688257/Schedules).
+ * - The provided test application provides a UDP client and server. If the UDP
+ * server is able to receive packets, the mechanism is considered to work correctly.
+ * You should also be able to ping the device from your host. See
+ * [tests/pkg_openwsn/README.md](../../tests/pkg_openwsn/README.md) for more details.
+ * - To speed up synchronization and make sniffing easier you can disable channel
+ * hopping by setting (`CFLAGS=-DIEEE802154E_SINGLE_CHANNEL=26`).
+ * - To sniff the packets either use a 802.15.4 capable board and follow
+ * [these instructions](https://github.com/RIOT-OS/applications/blob/master/sniffer/tools/README.md).
+ * Alternatively use a Raspberry Pi with an external radio such as Openlabs and
+ * incorporate Linux WPAN tools. In addition to that, there's also other
+ * hardware such as the ATUSB IEEE 802.15.4 USB Adapter which can directly be
+ * used on your Linux computer with WPAN tools installed. If you conduct your
+ * experiments on the IoT-LAB testbed you might want to use a:
+ * [sniffer profile](https://www.iot-lab.info/tutorials/radio-sniffer).
+ * - To explore the channel hopping mechanism there are rather expensive
+ * multi-channel sniffers such as the BeamLogic 802.15.4 Site Analyzer that can
+ * sniff all channels simultaneously. Alternatively you can set up multiple
+ * separate sniffer devices locally or make use of the `sniffer_aggregator` on
+ * the IoT-LAB testbed.
+ * - There is a collection of external tools to interact with the IoT nodes from
+ * a host computer via `openserial`. Please refer to the
+ * [OpenWSN software](https://github.com/openwsn-berkeley/openwsn-sw) repository
+ * for further information.
+ *
+ * ## Todos
+ *
+ * - `sctimer` to trigger an ISR immediately using sw isr.
+ * - `RTT_FREQUENCY` is not configurable for most platforms, implementations
+ *   should be adapted to make this configurable to be able to set RTT_FREQUENCY
+ *   to 32768Hz or the closer possible value.
```

Comment:
Yes, for most platforms the clock is 32Khz, but the define `RTT_FREQUENCY` is sometimes fixed at a value, if its too small (e.g. 1Hz) then the timer will not operate adequately enough even when shifted (since this is still in a trick to fake a "faster" timer at the cost of some tick inaccuracy by loosing the amount of bits you shift). And when you scale or shift there is still some cost, whether it is tick inaccuracy or conversion time, so when possible you would still want to set it to 32768. 

I've moved some platforms to be able to configure `RTT_FREQUENCY`, e.g. `stm32` and `nrf5x`, but I haven't done it for others like `sam0` (I'm just lucky that there the fixed value is 32768).